homeassistant/components/sensor/sytadin.py

```diff
@@ -0,0 +1,130 @@
+    OPTION_CONGESTION: ('Sytadin Congestion', '')
+}
+
+TIMEOUT = 10
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITION): vol.In(SENSOR_TYPES.keys()),
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_UPDATE_INTERVAL, default=timedelta(seconds=300)): (
+        vol.All(cv.time_period, cv.positive_timedelta)),
+})
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    options = [
+        it.strip() for it in config.get(CONF_MONITORED_CONDITION).split(',')
+    ]
+
+    name = config.get(CONF_NAME)
+
+    if OPTION_TRAFFIC_JAM in options:
+        if not name:
+            name = SENSOR_TYPES.get(OPTION_TRAFFIC_JAM)[0]
+        add_devices([
+            SytadinSensor(name, SYSTADIN, TRAFFIC_JAM_XPATH,
+                          TRAFFIC_JAM_REGEX,
+                          SENSOR_TYPES.get(OPTION_TRAFFIC_JAM)[1],
+                          config.get(CONF_UPDATE_INTERVAL))
+            ])
+    if OPTION_MEAN_VELOCITY in options:
+        if not name:
+            name = SENSOR_TYPES.get(OPTION_MEAN_VELOCITY)[0]
+        add_devices([
+            SytadinSensor(name, SYSTADIN,
+                          MEAN_VELOCITY_XPATH, MEAN_VELOCITY_REGEX,
+                          SENSOR_TYPES.get(OPTION_MEAN_VELOCITY)[1]+'/h',
+                          config.get(CONF_UPDATE_INTERVAL))
+            ])
+    if OPTION_CONGESTION in options:
+        if not name:
+            name = SENSOR_TYPES.get(OPTION_CONGESTION)[0]
+        add_devices([
+            SytadinSensor(name, SYSTADIN, CONGESTION_XPATH,
+                          CONGESTION_REGEX,
+                          SENSOR_TYPES.get(OPTION_CONGESTION)[1],
+                          config.get(CONF_UPDATE_INTERVAL))
+            ])
+
+
+class SytadinSensor(Entity):
+    """Sytadin Sensor."""
+
+    def __init__(self, name, url, xpath, regex, unit, interval):
+        """Initialize the sensor."""
+        self._state = None
+        self._name = name
+        self._url = url
+        self._xpath = xpath
+        self._regex = regex
+        self._unit = unit
+        self.update = Throttle(interval)(self._update)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    def _update(self):
+        """Fetch new state data for the sensor."""
+        import requests
+        import re
+        from lxml import etree
+
+        html = requests.get(self._url, timeout=TIMEOUT)
+        tree = etree.HTML(html.content)
+        extract_xpath = tree.xpath(self._xpath)
+
+        self._state = re.search(self._regex, extract_xpath[0]).group()
```

Comment:
I think that for maintenance reasons it would be a good idea to to take a look at [BeautifulSoup](https://www.crummy.com/software/BeautifulSoup/). Thus, we ca get rid of the `*_XPATH` constants and if the layout changes it could be easier to fix.

```python
import re
import requests
from bs4 import BeautifulSoup

REG = '(\d*\.\d+|\d+)'

raw_html = requests.get('http://www.sytadin.fr/sys/barometres_de_la_circulation.jsp.html').text
data = BeautifulSoup(raw_html, 'html.parser')

for entry in data.select('.barometre_valeur'):
    print(re.search(REG, entry.text).group())
```

Also, if there are three sensors in a user's configuration then three requests are made. Consider to use a data object and apply the throttle there or expose two values as attributes.

Reply:
copy that, maybe into next version

Reply:
`lxml` is around 4 MB and requires Cython while `BeautifulSoup` is 157 KB. I assume that we have requirements which have `lxml` as a dependency but evaluating it for a simple platform as this then it's overkill. Especially if we can do it in a simple way and simple is always better :wink: 

If you let the user configure the update interval we need to limit the requests. Thus with a data object that store the result. This will reduce the requests which are made by 60 %. 

Sure, include it in the commit and we can go on with the review.

Reply:
`lxml` is around 4 MB and requires Cython while `BeautifulSoup` is 157 KB. I assume that we have requirements which have `lxml` as a dependency but evaluating it for a simple platform as this then it's overkill. Especially if we can do it in a simple way and simple is always better :wink: 

If you let the user configure the update interval we need to limit the requests. Thus with a data object that store the result. This will reduce the requests which are made by 60 %. 

Sure, include it in the commit and we can go on with the review.